x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/search/aggregations/bucket/geogrid/GeoHexCellIdSource.java

```diff
@@ -76,10 +76,35 @@ private static class BoundedCellValues extends CellValues {
         private final boolean crossesDateline;
         private final GeoBoundingBox bbox;
 
+        private final long northPoleHex, southPoleHex;
+        private final boolean northPoleHexValid, southPoleHexValid;
+
         protected BoundedCellValues(MultiGeoPointValues geoValues, int precision, GeoBoundingBox bbox) {
             super(geoValues, precision);
             this.crossesDateline = bbox.right() < bbox.left();
             this.bbox = bbox;
+            northPoleHex = H3.geoToH3(90, 0, precision);
+            northPoleHexValid = northPoleValidHex(northPoleHex);
```

Comment:
I think it might be a little more consistent to call this in the validHex function. I'm not sure there is value in caching these results in the constructor.